fast_api_proj/app/main.py

```diff
@@ -0,0 +1,88 @@
+from fastapi import FastAPI, HTTPException, Depends
+from app.models.user import User, AdminUser, RegularUser
+from app.services.user_service import UserService, AdminUserService, RegularUserService
+from app.services.interfaces import IUserCreation, IUserRetrieval, IUserProfileUpdate
+import redis
+
+app = FastAPI()
+
+redis_client = redis.StrictRedis(host='localhost', port=6379, db=0, decode_responses=True)
+
+user_service = UserService(redis_client)
+admin_user_service = AdminUserService(redis_client)
+regular_user_service = RegularUserService(redis_client)
+
+
+def get_user_service() -> IUserCreation:
+    return user_service
+
+
+def get_admin_user_service() -> IUserCreation:
+    return admin_user_service
+
+
+def get_regular_user_service() -> IUserCreation:
+    return regular_user_service
+
+
+def get_user_retrieval_service() -> IUserRetrieval:
+    return user_service
+
+
+def get_user_profile_update_service() -> IUserProfileUpdate:
+    return user_service
```

Comment:
The above resources should be created in a separate file. They are implementation details which should not be coupled with the API page.